packages/common/src/services/FavoriteService.ts

```diff
@@ -15,31 +25,60 @@ export default class FavoriteService {
 
   private readonly apiService;
   private readonly storageService;
+  private readonly accountService;
 
-  constructor(apiService: ApiService, storageService: StorageService) {
+  constructor(@inject(INTEGRATION_TYPE) integrationType: string, apiService: ApiService, storageService: StorageService) {
     this.apiService = apiService;
     this.storageService = storageService;
+    this.accountService = getNamedModule(AccountService, integrationType, false);
   }
 
-  getFavorites = async (user: Customer | null, favoritesList: string) => {
-    const savedItems = user ? user.externalData?.favorites : await this.storageService.getItem<Favorite[]>(this.PERSIST_KEY_FAVORITES, true);
+  private validateFavorites(favorites: unknown) {
+    if (schema.validateSync(favorites)) {
+      return favorites as SerializedFavorite[];
+    }
+
+    return [];
+  }
+
+  private async getFavoritesFromAccount(user: Customer) {
+    const favorites = await this.accountService?.getFavorites({ user });
+
+    return this.validateFavorites(favorites);
+  }
 
-    if (savedItems?.length) {
-      const playlistItems = await this.apiService.getMediaByWatchlist(
-        favoritesList,
-        savedItems.map(({ mediaid }) => mediaid),
-      );
+  private async getFavoritesFromStorage() {
+    const favorites = await this.storageService.getItem(this.PERSIST_KEY_FAVORITES, true);
 
-      return (playlistItems || []).map((item) => this.createFavorite(item));
+    return this.validateFavorites(favorites);
+  }
+
+  getFavorites = async (user: Customer | null, favoritesList: string) => {
+    const savedItems = user ? await this.getFavoritesFromAccount(user) : await this.getFavoritesFromStorage();
+    const mediaIds = savedItems.map(({ mediaid }) => mediaid);
+
+    if (!mediaIds) {
+      return [];
     }
+
+    const playlistItems = await this.apiService.getMediaByWatchlist(favoritesList, mediaIds);
+
+    return (playlistItems || []).map((item) => this.createFavorite(item));
   };
 
   serializeFavorites = (favorites: Favorite[]): SerializedFavorite[] => {
     return favorites.map(({ mediaid }) => ({ mediaid }));
   };
 
-  persistFavorites = (favorites: Favorite[]) => {
-    this.storageService.setItem(this.PERSIST_KEY_FAVORITES, JSON.stringify(this.serializeFavorites(favorites)));
+  persistFavorites = async (user: Customer | null, favorites: Favorite[]) => {
```

Comment:
It is not an optional dependency but less important than the actual favorites. But I can swap the order!

I kept the entire customer object because we need to know what the integration service might need to save this information. However, I did notice that I didn't pass the customer to the account service...